client/gdc/singlecell.js

```diff
@@ -29,6 +33,99 @@ const columns = [
 	{ label: 'Sample Type' }
 ]
 
+export async function init(arg, holder, genomes) {
+	const plotAppApi = await appInit({
+		holder,
+		state: {
+			genome: gdcGenome,
+			dslabel: gdcDslabel,
+			termfilter: { filter0: arg.filter0 },
+			plots: [{ chartType: 'singleCellPlot' }]
+		},
+		noheader: true,
+		nobox: true,
+		hide_dsHandles: true,
+		genome: genomes[gdcGenome],
+		app: copyMerge(
+			{
+				getPlotHolder: (plot, div) => {
+					if (plot.chartType == 'gsea') {
+						const sandbox = newSandboxDiv(select(div.select(`#${plot.insertBefore}`).node().parentNode), {
+							close: () => {
+								plotAppApi.dispatch({
+									type: 'plot_delete',
+									id: plot.id
+								})
+							},
+							plotId: plot.id,
+							beforePlotId: plot.insertBefore || null,
+							style: {
+								width: '98.5%'
+							}
+						})
+						sandbox.header.text('GENE SET ENRICHMENT ANALYSIS')
+						return sandbox.body.style('white-space', 'nowrap')
+					} else {
+						return div.append('div')
+					}
+				}
+			},
+			arg.opts?.app || {}
+		),
+		singleCellPlot: arg.opts?.singleCell || {}
+	})
+	const api = {
+		update: async updateArg => {
+			if ('filter0' in updateArg) {
+				// the table and plots will need to be updated
+				//holder.selectAll('*').remove()
+
+				plotAppApi.dispatch({
+					type: 'app_refresh',
+					subactions: [
+						{
+							type: 'filter_replace',
+							filter0: updateArg.filter0
+						},
+						{
+							type: 'plot_edit',
+							id: plotAppApi.getState().plots[0].id,
+							config: { sample: undefined }
+						}
+					]
+				})
+
+				// const obj = {
+				// 	// old habit of wrapping everything
+				// 	errDiv: holder.append('div'),
+				// 	controlDiv: holder.append('div'),
+				// 	tableDiv: holder.append('div'),
+				// 	opts: {
+				// 		filter0: updateArg.filter0,
+				// 		experimentalStrategy: 'WXS'
+				// 	},
+				// 	api
+				// }
+				// makeControls(obj)
+				// await getFilesAndShowTable(obj); console.log(70, arg, updateArg)
+				// if (typeof callbacks?.postRender == 'function') {
+				// 	callbacks.postRender(publicApi)
+				// }
+			} else {
+				console.log(75, plotAppApi)
+				// the plots may change, but the table should not change
+				plotAppApi.dispatch({
+					type: 'plot_edit',
+					id: plotAppApi.getState().plots[0].id,
+					config: updateArg
+				})
+			}
+		}
+	}
+
+	return api
+}
+
 export async function gdcSinglecellUi({ holder, filter0, callbackOnRender, debugmode = false }) {
```

Comment:
looks like this func is only used in app.parseurl.js. i don't think it's essential to keep this `gdcsinglecell=1` url trigger, so we should delete this function and the trigger so no need to maintain this table rendering code